CRM/Event/Form/Registration/Confirm.php

```diff
@@ -1065,11 +1068,12 @@ public static function fixLocationFields(&$params, &$fields, &$form) {
       }
     }
 
-    // also add location name to the array
-    if ($form->_values['event']['is_monetary']) {
+    // Add the billing names to the billing address, if a billing name is set
+    if ($form->_values['event']['is_monetary'] && !empty($params['billing_first_name'])) {
```

Comment:
Yeah - I think it should go - but like I say - at min a comment to suggest that to the next person